### PR TITLE
Using docker cache with build of elm and add target to sample data

### PR DESCRIPTION
**Pull request by gomex**

### CODE REVIEW

**Comment by gomex** [comment]
We need to merge #121 first

**Comment by gomex** [comment]
Something breaks :( 

When I run npm run watch I got this:

```
# npm run watch
npm info it worked if it ends with ok
npm info using npm@3.10.8
npm info using node@v6.9.1
npm info lifecycle @~prewatch: @
npm info lifecycle @~watch: @

> @ watch /
> gulp watch

[07:21:12] Using gulpfile /gulpfile.js
[07:21:12] Starting 'elm'...
Success! Compiled 167 modules.                                      
Successfully generated /117015-17-cn07og.ldcsl6usor.js
[07:21:47] Finished 'elm' after 35 s
[07:21:47] Starting 'watch'...
[07:21:47] Finished 'watch' after 12 ms
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: EPERM: operation not permitted, stat '/proc/1/map_files/400000-4f2000'
    at Error (native)

npm info lifecycle @~watch: Failed to exec watch script
npm ERR! Linux 4.4.41-moby
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "watch"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! @ watch: `gulp watch`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ watch script 'gulp watch'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp watch
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs 
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls 
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /npm-debug.log
```

File: README.md [hunk 1]
```diff
@@ -135,23 +135,33 @@ There is also a [tapioca-wrapper](https://github.com/vintasoftware/tapioca-wrapp
...
+or
+
+```console
+docker-compose up -d --build
+docker-compose run --rm jarbas python manage.py migrate
+docker-compose run --rm jarbas python manage.py ceapdatasets
+docker-compose run --rm jarbas python manage.py collectstatic --no-input
 ```
 
 You can access it at [`localhost:80`](http://localhost:80/). However your database starts empty and you still have to collect your static files:
```
Comment (cuducos) [inline README.md]: Three things here:

* As we are extracting _variables_ for development and production environment, is it easy to make `make run.devel` to be available in `localhost:8000` (Django's default)?
* The sentence makes no sense anymore, since `collectstatic` happened in the previous step. 
* Maybe rephrase it to explain that *either* one can load the sample data from the repo just to get started, **or** run Rosie to get the full dataset (maybe moving a sentence that is a few lines below to this point).

Comment (gomex) [inline README.md]: > As we are extracting variables for development and production environment, is it easy to make make run.devel to be available in localhost:8000 (Django's default)?

It is easy! We just need to change docker-compose.yml (development file)

> The sentence makes no sense anymore, since collectstatic happened in the previous step.
> Maybe rephrase it to explain that either one can load the sample data from the repo just to get started, or run Rosie to get the full dataset (maybe moving a sentence that is a few lines below to this point).

I agree. I will change it! :)

File: docker-compose.yml [hunk 2]
```diff
@@ -23,20 +23,20 @@ services:
...
       - "./:/code"
+      - "assets:/code/jarbas/frontend/static"
     command: [npm, run, watch]
   nginx:
     build:
       context: .
       dockerfile: Dockerfile-nginx
     volumes:
-      - "staticfiles:/opt/jarbas/staticfiles"
+      - "assets:/opt/jarbas/staticfiles"
```
Comment (petriuslima) [inline docker-compose.yml]: Sorry, I don't know about nginx, is this line to serve staticfiles from nginx instead of jarbas container?
Are the files from jarbas container still being served?

Comment (gomex) [inline docker-compose.yml]: my idea is don't need anymore the collectstatic from django :)

**Comment by gomex** [comment]
@cuducos now you can merge it :)

**Comment by cuducos** [comment]
Testing now, probably merging in the next few minutes, yay 🎉

**Comment by cuducos** [comment]
I got an error here, nginx is not serving the static files:

```console
$ curl -I http://localhost:8000/static/app.js

HTTP/1.1 404 Not Found
Server: nginx/1.11.5
Date: Mon, 16 Jan 2017 14:26:26 GMT
Content-Type: text/html
Content-Length: 169
Connection: keep-alive
```

Is it just me?

**Comment by cuducos** [comment]
Some extra debugging comments that might be helpful:

Files inside the nginx containers are messed up. In a local environment there are two steps to be considered:

1. `yarn assets` generates `jarbas/frontend/statis/app.js`
2. `python manage.py collectstatic` copies `jarbas/frontend/statis/app.js` together with all other static files (images, favicons, files from Django REST Framework and Django Admin) to `staticfiles/`

So in the end this is the structure expected for the `staticfiles/` that nginx is going to serve:

```console
$ ls staticfiles
admin              app.js             ceap-datasets.html digitalocean.png   rest_framework
```

But that's not what we have within the nginx container:

```console
$ docker-compose run --rm nginx /bin/bash
root@173876fe17e2:~# ls /opt/jarbas/staticfiles/
ceap-datasets.html  jarbas
root@173876fe17e2:~# ls /opt/jarbas/staticfiles/jarbas/frontend/static/
app.js	ceap-datasets.html  digitalocean.png
```

At `/opt/jarbas/staticfiles` (ie what nginx is serving) there are two things:

1. `ceap-datasets.html` (as expected, synce `python manage.py ceapdatasets` places it exactly there)
2. the project directory `jarbas/` (but I don't know why, anyway I consider this is a bug)

And we're missing the more importante: the collected static files (`admin`, `app.js`,  `digitalocean.png`, `rest_framework`…).

**Comment by cuducos** [comment]
Just sent [a PR to `gomex:master`](https://github.com/gomex/jarbas/pull/1) fixing [the bug describes earlier](https://github.com/datasciencebr/jarbas/pull/127#pullrequestreview-16835491), does it make sense, @gomex?

**Comment by gomex** [comment]
I can't reproduce your error cuducos :(

```
$ make run.devel
$ docker-compose exec nginx bash
# ls /opt/jarbas/staticfiles/
app.js              ceap-datasets.html  digitalocean.png    favicon/  
```

Did you removed all containers and volumes?

```
docker-compose rm -v
```

To discovery what volumes to remove:

```
docker volume ls | more
```

To remove volumes:

```
docker volume rm <volume1> <volume2> <volume3>
```

**Comment by cuducos** [comment]
Ooops… my bad. Docker noob strikes again.

Cleaning the volumes worked… merging it 🎉

**Comment by gomex** [comment]
THANKS!!!! \o/
